inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/random_uniform/random_uniform_kernel_ref.cpp

```diff
@@ -0,0 +1,116 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "random_uniform_kernel_ref.h"
+
+#include <kernel_selector_utils.h>
+
+namespace {
+
+using namespace kernel_selector;
+
+
+size_t GetFullShapeSize(const Tensor::DataTensor &tensor) {
+    return tensor.Batch().v * tensor.Feature().v * tensor.Z().v * tensor.Y().v * tensor.X().v * tensor.W().v;
+}
+
+int getStep(const random_uniform_params &params) {
+    return BytesPerElement(params.output.GetDType()) > 4 ? 2 : 4;
+}
+
+size_t GetGwsSize(const random_uniform_params &params) {
+    size_t shapeSize = GetFullShapeSize(params.output);
+    int step = getStep(params);
+    return (shapeSize >> (step >> 1)) + static_cast<int>(shapeSize % step > 0);
+}
+
+CommonDispatchData SetDefault(const random_uniform_params &params, const optional_params &) {
+    CommonDispatchData dispatchData;
+    dispatchData.gws = {GetGwsSize(params), 1, 1};
+    dispatchData.lws = GetOptimalLocalWorkGroupSizes(dispatchData.gws, params.engineInfo);
+    return dispatchData;
+}
+
+}  // namespace
+
+namespace kernel_selector {
+
+JitConstants RandomUniformKernelRef::GetJitConstants(const random_uniform_params &params) const {
+    JitConstants jit = MakeBaseParamsJitConstants(params);
+    // When both seed values are equal to zero RandomUniform should generate non-deterministic sequence.
+    if (params.global_seed == 0 && params.op_seed == 0) {
+        std::srand(std::time(nullptr));
+        jit.AddConstant(MakeJitConstant("GLOBAL_SEED", std::rand()));
+    } else {
+        jit.AddConstant(MakeJitConstant("GLOBAL_SEED", params.global_seed));
+    }
+
+    jit.AddConstant(MakeJitConstant("OP_SEED", params.op_seed));
+    jit.AddConstant(MakeJitConstant("OUTPUT_SHAPE_SIZE", GetFullShapeSize(params.output)));
+    jit.AddConstant(MakeJitConstant("OUTPUT_STEP", getStep(params)));
+    return jit;
+}
+
+
+KernelsData RandomUniformKernelRef::GetKernelsData(const Params &params, const optional_params &options) const {
+    if (!Validate(params, options)) {
+        return {};
+    }
+
+    KernelData kernel_data = KernelData::Default<random_uniform_params>(params);
+    const random_uniform_params &new_params = dynamic_cast<const random_uniform_params &>(*kernel_data.params.get());
+
+    auto dispatch_data = SetDefault(new_params, options);
+    auto entry_point = GetEntryPoint(kernelName, new_params.layerID, params, options);
+
+    auto random_uniform_specific_jit = GetJitConstants(new_params);
+    auto jit = CreateJit(kernelName, random_uniform_specific_jit, entry_point);
+
+    FillCLKernelData(kernel_data.kernels[0], dispatch_data, params.engineInfo, kernelName, jit, entry_point, "", false,
+                     false, 3);
+
+    return {kernel_data};
```

Comment:
`KernelData` is large stucture and you are making copy of it here. actually, you are copying it twice in this line: once to `std::initializer_list<KernelData>` and once from it to `vector`. you can avoid copy from `initializer_list` by replacing initializer list constructor with `push_back()` and you can avoid copy from `kernel_data` by moving(or by pushing back result of .`KernelData::Default` directly)